### PR TITLE
Add centralized entry point to importer [Do not Merge]

### PR DESCRIPTION
**Pull request by vfonic**
* This can be used from anywhere to kick-off a new import: rake task, cli, background job
* Change `ShopifyImport::Base` instance methods `count` and `all` to be
  class methods.
  * This makes it have the same API to activerecord module methods
    `where`, `find`, `all`, `count`, etc.
  * https://github.com/rails/rails/blob/f7388635f246b97a1aa0be54579339c6659fb865/activerecord/lib/active_record/relation/finder_methods.rb#L64
  * https://github.com/rails/rails/blob/f7388635f246b97a1aa0be54579339c6659fb865/activerecord/lib/active_record/scoping/named.rb#L24
  * https://github.com/rails/rails/blob/f7388635f246b97a1aa0be54579339c6659fb865/activerecord/lib/active_record/relation/calculations.rb#L39
  * https://github.com/rails/rails/blob/f7388635f246b97a1aa0be54579339c6659fb865/activerecord/lib/active_record/relation/query_methods.rb#L599
* Don't pass credentials to `ShopifyImport::Base` and subclasses
  * We initialize session globally, so there's no need to pass it to
    have the initiated session
* Add `authenticate_with_shopify` spec helper method
  * Mainstream the way we mock connecting to Shopfiy in the specs
  * This also hides making the connection in the specs where acting with
    the connection is not the main focus of the spec
* Remove `ShopifyImport::Base` specs
  * Creating a connection to the client functionality is moved to
    `ShopifyImport::Invoker` therefore these specs can be (re)moved there
  * Calling `.count` on `ShopifyImport::Base` should raise a name error
    not `ActiveResource::ResourceNotFound`; name error comes from trying
    to make an API call `ShopifyAPI::Base.count`; I actually started receiving
    this error when I changed `count` from instance to class method
  * `ShopifyImport::Base` functionality should really be tested through
    its subclasses. We never want to instantiate `ShopifyImport::Base`
    so, if we test that `ShopifyImport::Product` behaves like how we want,
    there's no need to test `ShopifyImport::Base` separately. Test the
    functionality, not for higher test coverage.

NB: Invoker is heavily used for "integration testing project": https://github.com/vfonic/testing_spree_shopify_importer

### CODE REVIEW

**Comment by vfonic** [comment]
Added `[Do not merge]` as this PR is not compatible with https://github.com/spark-solutions/spree_shopify_importer/pull/28
I think #28 can be merged sooner, so when that's done, I'll rebase this PR on top of those changes and make it ready for merging.

File: app/services/shopify_import/importers/base_importer.rb [hunk 1]
```diff
@@ -1,16 +1,7 @@
 module ShopifyImport
   module Importers
     class BaseImporter
-      def self.import!(**params)
```
Comment (Argonus) [inline app/services/shopify_import/importers/base_importer.rb]: @vfonic I think that we should have params here. Not only for credentials but also for filters like `created_at_min` etc.

File: app/services/shopify_import/importers/products_importer.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ class ProductsImporter < BaseImporter
       private
 
       def resources
-        ShopifyImport::Product.new(credentials: @credentials).find_all(@params)
```
Comment (Argonus) [inline app/services/shopify_import/importers/products_importer.rb]: @vfonic as above.

File: app/services/shopify_import/importers/base_importer.rb (hunk 1 above)
Comment (vfonic) [inline app/services/shopify_import/importers/base_importer.rb]: Let's add those when we actually use them? Right now I see we're adding a lot of functionality that we _might_ use, like base.new with both client and credentials parameters, params/options passing around, etc.

I agree we should think about future needs, like re-importing and importing from multiple stores, but I don't think we should focus on providing functionality for that step right now. Having DataStore where we store what has been added is good enough. From there we can build on and have other extra functionalities. So I'd first focus on having "one full import working end-to-end". Adding params here later on will be extremely easy.

Comment (vfonic) [inline app/services/shopify_import/importers/base_importer.rb]: @Argonus added back the params

**Comment by vfonic** [comment]
@Argonus I think this PR can also be merged, but please **don't** merge both this and #30! Choose one and I'll then rebase the other on top of the one you merged

**Pull request closed without merging**